kiro-synth-core/src/effects/delay.rs

```diff
@@ -0,0 +1,77 @@
+use crate::float::Float;
+use std::collections::VecDeque;
+
+struct DelayLine<F: Float> {
+  sample_rate: F,
+  buffer: VecDeque<F>,
+}
+
+impl<F: Float> DelayLine<F> {
+  pub fn new(sample_rate: F) -> Self {
+    let delay_max = F::val(2.0);
+    let len = (sample_rate * delay_max).ceil().to_usize().unwrap();
+    Self {
+      sample_rate,
+      buffer: VecDeque::<F>::with_capacity(len),
+    }
+  }
+
+  pub fn update(&mut self, input: F) {
+    self.buffer.pop_front();
+    self.buffer.push_back(input);
+  }
+
+  pub fn get(&self, index: F) -> F {
```

Comment:
the unit for index is `seconds` right ? can we change the name to avoid confusion with an actual index in the buffer ?

Reply:
We could rename it with `time` for example